### ISSUE TITLE
Authorize an Org stopped working in Codespaces

### ISSUE THREAD

**Issue by eburtness** [forcedotcom/salesforcedx-vscode#5704]
### Summary
We're using vscode extensions in GitHub codespaces and today started getting an error when trying to authorize an org.

Output tab gives command as `sfdx org:login:device --alias Dev03 --instance-url https://test.salesforce.com --set-default --json --loglevel fatal` and when I run that command from the terminal it works just fine. The error only happens when I try to auth via "SFDX: Authorize an Org".

### Steps To Reproduce:
1.  From command palette, choose **SFDX: Authorize an Org**
2. Choose Production or Sandbox
3. Enter an alias
4. In the new browser tab, click **Connect**
5. Log in to your desired org
6. On the "Allow Access?" screen, click on **Allow**

### Expected result
Success message

### Actual result
Error message:

> We can't authorize you because of an OAuth error. For more information, contact your Salesforce administrator.
> 
> OAUTH_APPROVAL_ERROR_GENERIC : An unexpected error has occured during authentication. Please try again.

URL query string is `error=invalid_grant&error_description=expired+authorization+code`

### Additional information

**Salesforce Extension Version in VS Code**: 61.6.0
**Salesforce CLI Version**: @salesforce/cli/2.52.6 linux-x64 node-v20.15.1
**OS and version**: Linux Ubuntu 22.04.1
**VS Code version**: 1.91.1
**Most recent version of the extensions where this was working**: 61.6.0

To clarify:
- I'm not sure it was working with 61.6.0. I tried 61.5.0, 59.17.0, and a few others in between, and none of them are working today. But we had no problems until today, so I doubt this was caused simply by a change in the extension.
- I also tried version 2.44.8 and 2.49.7 of the CLI, and got the same error.
- I tried in Chrome, Edge, and Safari, with the same results.
- On my local (Mac) installation of VS Code (with the same CLI and extension versions) the same process works fine. I'm only seeing the error from a codespace using "SFDX: Authorize an Org".
- I'm seeing the same error with sandboxes and my Production org

**Comment by eburtness** [comment]
Hi @svc-idee-bot I've updated my description to better match the example format, and yes I am still seeing the issue with the latest version of the Salesforce Extension Pack.

**Comment by peternhale** [comment]
@eburtness The cause of this issue is being tracked by another Salesforce team.

The workaround, as you discovered, will be to run the auth command in a terminal in your hosted environment.

**Comment by diyer** [comment]
@eburtness you can follow this known issue on this thread https://github.com/forcedotcom/code-builder-feedback/wiki/Known-Gaps-and-Issues#oauth_approval_error_generic-authentication-error

**Comment by jonnyhork** [comment]
@eburtness If you don't mind sharing, I am curious about your use case for SFDX in Codespaces. Why did you choose to use Codespaces in the first place? Are you using the [salesforce/cli docker image](https://hub.docker.com/r/salesforce/cli/tags) as a base for you codespace? Have you considered or would use Code Builder instead?

**Comment by eburtness** [comment]
Hi @jonnyhork, happy to share!

We started using Codespaces to have a development environment that I could configure and then anyone on my team could use, with the configuration living right in the repo. It avoids getting IT involved to set up VS Code, a JDK, the CLI, Git, etc. every time someone gets a new or refreshed laptop.

Yes we're using the `salesforce/cli` docker image.

I tried Code Builder when it was in beta. Some advantages I saw for Codespaces over Code Builder at that time:
- Can automatically install any VS Code extension via devcontainer.json
- Can automatically install sf plugins (and do other configuration) via Dockerfile
- Can use a local installation of VS Code as the client, instead of a browser window
- Integration with GitHub — you just launch it from GitHub and the repo is all cloned and ready to go (I don't remember Code Builder's experience for cloning the repo, maybe it's not bad)
- Codespaces have been reliable and trouble-free, and the performance is good. During beta, Code Builder would frequently lose connection and need a full restart (like every few minutes). I expect that was resolved for GA, but I haven't tested to see.

I would like to try Code Builder again, I just haven't bothered yet because Codespaces have been working great for us.
